.github/workflows/lint-changed-files.yaml

```diff
@@ -19,20 +19,16 @@ jobs:
 
       - uses: r-lib/actions/setup-r-dependencies@v2
         with:
-          extra-packages: |
+          packages: |
             any::gh
             any::lintr
             any::purrr
-          needs: check
```

Comment:
the default version of this GHA installs the package, along with its dependencies; that doesn't seem necessary and takes a lot of time, so I'm removing those steps and just installing the bare minimum to run `{lintr}`